static/js/emoji.js

```diff
@@ -18,6 +18,36 @@ var zulip_emoji = {
     deactivated: false,
 };
 
+// Regular Expressions (as strings, not RegEx Objects) of emoticons, and which
+// emoji they should become. Duplicates are allowed, such as for very different
+// emoticons that should still become the same emoji (although a RegEx could
+// also be used).
+var EMOTICON_CONVERSIONS = {
+    // Matches ":)", ": )", and ":-)".
+    ':( |-)?\\)': '😃',
+    // Matches "<3".
+    '<3': '❤️',
+    // Matches ";)", "; )", and ";-)".
+    ';( |-)?\\)': '😉',
+    // Matches ":(", ": (", and ":-(".
+    ':( |-)?\\(': '🙁',
+    // Matches ":,(", ":, (", ":,-(", ":'(", ":' (", and ":'-(".
+    ':(,|\')( |-)?\\(': '😢',
+    // Matches ":O", ": O", ":-O", ":0", ": 0", and ":-0".
```

Comment:
I would like to raise a point against this conversion. It is likely to have code snippets that have `: 0` as a part of them. On the whole, I do not see why we should translated emojis when the text has spaces in between. `: )` might be someone deliberately trying to avoid such a conversion.

Reply:
Yeah, I agree; I think we should try to make this list minimal, to avoid accidental use of emoticons.  Avoiding accidental use of syntax is one of the philosophical principles in Zulip's message formatting, and it seems like we should stick with that here.

Reply:
@timabbott OK, I removed the space option in the RegExes. Please ket me know if there is more I should fix or if this could be merged. Thank you!

Reply:
(also @aero31aero)